src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -8411,6 +8411,104 @@ public static int Main()
+    {
+    }
+}
+";
+
+            var comp = CreateStandardCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+            var typeMember = comp.GetMember<NamedTypeSymbol>("System.Runtime.CompilerServices.SkipLocalsInitAttribute");
+
+            Assert.Equal(wellKnownType, typeMember);
+        }
+
+        [Fact]
+        public void SkipLocalsInitAttributeRecognizedAsWellKnownTypeWhenDeclaredInMetadata()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var metadataComp = CreateStandardCompilation(source);
+            metadataComp.VerifyDiagnostics();
+
+            var comp = CreateStandardCompilation("", references: new[] { metadataComp.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+            var typeMember = comp.GetMember<NamedTypeSymbol>("System.Runtime.CompilerServices.SkipLocalsInitAttribute");
+
+            Assert.Equal(wellKnownType, typeMember);
+        }
+
+        [Fact]
+        public void SkipLocalsInitAttributeFromSourceRecognizedAsWellKnownTypeWhenDeclaredInSourceAndMetadata()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var metadataComp = CreateStandardCompilation(source);
+            metadataComp.VerifyDiagnostics();
+
+            var comp = CreateStandardCompilation(source, references: new[] { metadataComp.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+            
+            Assert.IsType<SourceNamedTypeSymbol>(wellKnownType);
+        }
+
+        [Fact]
+        public void SkipLocalsInitAttributeMissingWhenDeclaredMoreThanOnceInOtherAssemblies()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var metadataComp1 = CreateStandardCompilation(source);
+            metadataComp1.VerifyDiagnostics();
+
+            var metadataComp2 = CreateStandardCompilation(source);
+            metadataComp2.VerifyDiagnostics();
+
+            var comp = CreateStandardCompilation("", references: new[] { metadataComp1.EmitToImageReference(), metadataComp2.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+
+            Assert.True(wellKnownType.IsErrorType());
```

Comment:
Why did we pick error type here vs. just picking one of them to win? Seems like this is similar to the extension attribute where we pick one. Perhaps I'm missing something here @agocke

Reply:
That's the default behavior for Well Known Types after C# 7. Before C# 7 the first candidate would be returned with a warning. Do you think it's better to reconsider this behavior?

Reply:
Yeah, the doc comment mentions that we changed this behavior post-C# 7. cc @jcouv who may have more background here, since it seemed pointed at ValueTuple.